[DebugInfo][RemoveDIs] Rip out the UseNewDbgInfoFormat flag

jmorse opened the pull request:
Start removing debug intrinsics support -- starting with the flag that controls production of their replacement, debug records. This patch removes the command-line-flag and with it the ability to switch back to intrinsics. The module / function / block level "IsNewDbgInfoFormat" flags get hardcoded to true, I'll to incrementally remove things that depend on those flags.

jmorse committed:
[DebugInfo][RemoveDIs] Rip out the UseNewDbgInfoFormat flag

Start removing debug intrinsics support -- starting with the flag that
controls production of their replacement, debug records. This patch removes
the command-line-flag and with it the ability to switch back to intrinsics.
The module / function / block level "IsNewDbgInfoFormat" flags get
hardcoded to true, I'll to incrementally remove things that depend on those
flags.

OCHyams reviewed (commented):
Mostly LGTM, couple questions inline (plus pls appease the format bot)

jmorse committed:
Remove MLIR uses of the old-dbg-info-mode flag

jmorse committed:
clang-format

jmorse committed:
Comment on future users of conversion routines

jmorse committed:
Remove a now redundant test

jmorse commented on llvm/lib/IR/IRPrintingPasses.cpp:
```diff
@@ -42,12 +40,10 @@ class PrintModulePassWrapper : public ModulePass {
 
   bool runOnModule(Module &M) override {
-    ScopedDbgInfoFormatSetter FormatSetter(M, UseNewDbgInfoFormat);
+    ScopedDbgInfoFormatSetter FormatSetter(M, true);
```
It's being eliminated in the next commit to send up, figured I'd make them incremental.

jmorse commented on llvm/unittests/IR/DebugInfoTest.cpp:
```diff
@@ -954,6 +931,12 @@ TEST(MetadataTest, ConvertDbgToDbgVariableRecord) {
+  // The IR above will be autoupgraded to debug records: but this test is about
+  // the conversion routines, so convert it back. This test will have value
+  // going forwards as we're likely to have a few corner cases where we want to
+  // use intrinsics. 
```
Added comment about how DXIL is likely to want this to keep working, thus we need to keep testing it.

jmorse commented on llvm/unittests/IR/ValueTest.cpp:
```diff
@@ -302,6 +298,7 @@ TEST(ValueTest, replaceUsesOutsideBlock) {
   auto GetNext = [](auto *I) { return &*++I->getIterator(); };
 
   Function *F = M->getFunction("f");
+  F->convertFromNewDbgValues();
```
Ah, hadn't spotted the nearby DbgRecord implementation -- I've deleted the intrinsic one.

OCHyams reviewed (approved):
LGTM

jmorse committed:
Remove a RUNline with experimental-debuginfo-iterators

This was adding extra coverage in the past; now it's un-necessary.

jmorse committed:
Remove some RUNlines from a flang test

This flang has been a dummy for roughly a month now -- the test coverage
through the extra RUN lines made sense in the past, but now it's redundant.

jmorse commented:
Removing the `--experimental-debuginfo-iterators` flag from 1 flang and 1 mlir test too -- the flag has been a dummy for a month and doesn't do anything, the extra RUNlines were just extra coverage for when intrinsic support was still feasible.
